### DISCUSSION TITLE
Polyline performance

### DISCUSSION THREAD

**Discussion by t-h-f** [react-native-maps/react-native-maps#4438] [Q&A]
Please can someone clarify the correct approach to drawing lines between each coordinate that describes a journey on the map from point A to point B? Example: I have 400 coordinates saved for a small journey of 2.5 miles along various roads from point A to B and when I render a Polyline passing in 400 coordinates it takes several seconds to the Polyline to render, which is not ideal performance.

**Comment by johnmuller76** [comment]
You can use [simplify-js](https://mourner.github.io/simplify-js/) library to reduce number of points to be passed to polyline, this reduces load on JS bridge, which I believe slows things down for large number of coordinates.

**Reply by webraptor** [reply]
> You can use [simplify-js](https://mourner.github.io/simplify-js/) library to reduce number of points to be passed to polyline, this reduces load on JS bridge, which I believe slows things down for large number of coordinates.

simplify-js improves performance a lot.
